### PR TITLE
Services: API overhaul for using only ids

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
Finishes the conversion so that Services look similar to all other API's. That is, submit a list of services, get the id of the result back (in this case a procedures id).

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by dgasmith** [comment]
@Lnaden @loriab Can I get another review on the last few changes?

File: qcfractal/interface/collections/collection.py [hunk 1]
```diff
@@ -57,12 +57,6 @@ class DataModel(BaseModel):
         provenance: Dict[str, str] = {}
         id: str = 'local'
 
-        class Config:
```
Comment (Lnaden) [inline qcfractal/interface/collections/collection.py]: Did you want datamodel to be mutable?

File: qcfractal/interface/collections/openffworkflow.py [hunk 2]
```diff
@@ -85,14 +85,14 @@ class DataModel(Collection.DataModel):
             }
         }
         torsiondrive_static_options: Dict[str, Any] = {
-            "torsiondrive_meta": {},
-            "optimization_meta": {
+            "keywords": {},
```
Comment (Lnaden) [inline qcfractal/interface/collections/openffworkflow.py]: This is fine, more from a user perspective, do we have anyone we need to tell about this change?

File: qcfractal/interface/models/gridoptimization.py [hunk 3]
```diff
@@ -53,26 +53,20 @@ def check_step_type(cls, v):
...
-
     class Config:
+        extra = "forbid"
         allow_mutation = False
 
 
-class OptOptions(BaseModel):
+class GOKeywords(BaseModel):
     """
     GridOptimization options
```
Comment (Lnaden) [inline qcfractal/interface/models/gridoptimization.py]: This docstring now mimics the docstring in OptimizationSpecification, one of them should be changed to better match the true description

File: qcfractal/interface/collections/openffworkflow.py (hunk 2 above)
Comment (dgasmith) [reply qcfractal/interface/collections/openffworkflow.py]: Not yet, no.

File: qcfractal/interface/models/gridoptimization.py (hunk 3 above)
Comment (dgasmith) [reply qcfractal/interface/models/gridoptimization.py]: Not quite `OptimizationSpecification` includes Keywords and other things. The `GOKeywords` are explicitly just keywords.

File: qcfractal/interface/collections/collection.py (hunk 1 above)
Comment (dgasmith) [reply qcfractal/interface/collections/collection.py]: This is never called so its attributes do not matter as Config is no inherited. But in general, yes the datamodel should be mutable.

**Pull request merged**